airflow/dags/rt_loader/create_validation_params.py

```diff
@@ -0,0 +1,57 @@
+# ---
+# python_callable: main
+# provide_context: true
+# dependencies:
+#   - external_calitp_validation_params
+# external_dependencies:
+#   - gtfs_loader: calitp_feed_status
+# ---
+
+from calitp import query_sql, save_to_gcfs
+from calitp.config import get_bucket
+
+
+def main(execution_date, **kwargs):
+    date_string = execution_date.to_date_string()
+
+    raw_params = query_sql(
+        f"""
+        SELECT
+            calitp_itp_id,
+            calitp_url_number,
+            calitp_extracted_at
+        FROM gtfs_schedule_history.calitp_feed_status
+        WHERE
+            is_extract_success
+            AND NOT is_parse_error
+            AND calitp_extracted_at = "{date_string}"
+        """,
+        as_df=True,
+    )
+
+    # Note that raw RT data is currently stored in the production bucket,
+    # and not copied to the staging bucket
+    prefix_path_schedule = f"{get_bucket()}/schedule/{execution_date}"
+
+    # This prefix limits the validation to only 1 hour of data currently
+    prefix_path_rt = f"gtfs-data/rt/{date_string}T00:*"
+
+    params = raw_params.assign(
```

Comment:
These params need to be expanded to be on a per-feed, per-rt-feed-type basis, otherwise, the rt validator won't work properly.

The RT validator notes that it automatically sorts files based on file timestamp ([see docs on `-sort`](https://github.com/CUTR-at-USF/gtfs-realtime-validator/tree/master/gtfs-realtime-validator-lib#command-line-config-parameters)). This is important because some validation rules are dependent on analyzing the previous chronological file. In order to accomplish this though, there seems to be an assumption that the files being analyzed are all of the same type, ie they're all trip updates, all vehicle positions or all service alerts. So, this code currently jumbles all these file types together which will result in validations of different file types which may incorrectly trigger (or not trigger) a validation rule. From the logs posted earlier it seems like the sorting is already occurring correctly, so that's good.

So, please update these params to grab only one specific rt file type at a time in the glob path.

This should actually provide more opportunities for parallelization than before.

Reply:
Nice catch, thanks! I will rework these as well as the output formats. I think it may also be worth splitting up the validations into separate tasks like the loading, so  the DAG structure is a bit cleaner, too.

Reply:
This is done